odoo/tests/common.py

```diff
@@ -415,18 +415,20 @@ def with_user(self, login):
     @contextmanager
     def _assertRaises(self, exception, *, msg=None):
         """ Context manager that clears the environment upon failure. """
-        with super(BaseCase, self).assertRaises(exception, msg=msg) as cm:
+        with super().assertRaises(exception, msg=msg) as cm, ExitStack() as ex:
             if hasattr(self, 'env'):
-                with self.env.cr.savepoint():
+                try:
+                    ex.enter_context(self.env.cr.savepoint())
                     if issubclass(exception, AccessError):
                         # The savepoint() above calls flush(), which leaves the
                         # record cache with lots of data.  This can prevent
                         # access errors to be detected. In order to avoid this
                         # issue, we clear the cache before proceeding.
                         self.env.cr.clear()
-                    yield cm
-            else:
-                yield cm
+                except exception:
+                    _logger.exception("%r raised during assertRaises setup")
```

Comment:
Wait, how does `%r` work in there?

Reply:
It doesn't, that's a leftover from a previous version of the code (where it was `_logger.error`) and I missed it, good eye.